matching/src/main/scala/org/kframework/backend/llvm/matching/Matrix.scala

```diff
@@ -46,10 +46,11 @@ class Column(val fringe: Fringe, val patterns: IndexedSeq[Pattern[String]], val
       Double.PositiveInfinity
     } else {
       var result = 0.0
+      val heuristic = new FHeuristic()
       for (i <- patterns.indices) {
         if (clauses(i).action.priority != clauses.head.action.priority)
           return withChoice(result)
-        result += patterns(i).score(fringe, clauses(i), key, isEmpty)
+        result += patterns(i).score(heuristic, fringe, clauses(i), key, isEmpty)
```

Comment:
if you make it an object instead of a class, `heuristic` can just become `DefaultHeuristic`.